CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/utility/SelectAllPluginNGTest.java

```diff
@@ -0,0 +1,328 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.views.dataaccess.plugins.utility;
+
+import au.gov.asd.tac.constellation.graph.Graph;
+import au.gov.asd.tac.constellation.graph.ReadableGraph;
+import au.gov.asd.tac.constellation.graph.WritableGraph;
+import au.gov.asd.tac.constellation.graph.locking.DualGraph;
+import au.gov.asd.tac.constellation.graph.schema.BareSchemaFactory;
+import au.gov.asd.tac.constellation.graph.schema.SchemaFactoryUtilities;
+import au.gov.asd.tac.constellation.graph.schema.visual.VisualSchemaFactory;
+import au.gov.asd.tac.constellation.graph.schema.visual.concept.VisualConcept;
+import au.gov.asd.tac.constellation.plugins.PluginException;
+import au.gov.asd.tac.constellation.plugins.PluginExecution;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * Cases of Edit tested: empty graph with visual schema, empty graph with bare
+ * schema, empty graph visual schema with attributes, visual graph with two vx
+ * and attributes, visual graph with two vx and two tx.
+ *
+ *
+ * @author aldebaran30701
+ */
+public class SelectAllPluginNGTest {
+
+    private Graph graph;
+    private int selectedV, selectedT;
+    private int vxId1, vxId2, txId1, txId2;
+
+    public SelectAllPluginNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of edit method, of class SelectAllPlugin. No attributes added so
+     * should throw a pluginexception
+     */
+    @Test(expectedExceptions = PluginException.class)
+    public void testEdit() throws Exception {
+        System.out.println("select all edit with nothing on graph");
+        // Open a new graph
+        graph = new DualGraph(SchemaFactoryUtilities.getSchemaFactory(VisualSchemaFactory.VISUAL_SCHEMA_ID).createSchema());
+        int vxCount = 0;
+        int txCount = 0;
+
+        // run select all plugin
+        PluginExecution.withPlugin(new SelectAllPlugin()).executeNow(graph);
+
+        // Verify no extra elements are added
+        final ReadableGraph rg = graph.getReadableGraph();
```

Comment:
I could be wrong, but if you're expecting the PluginException to be thrown (and you're not catching it in the test), I would've thought the execution of the test doesn't go beyond the PluginExecution line

Reply:
Good point, It wasn't getting reached.